pkg/registry/softwarecomposition/applicationprofile/strategy.go

```diff
@@ -65,7 +65,9 @@ func (applicationProfileStrategy) PrepareForUpdate(ctx context.Context, obj, old
 	oldAP := old.(*softwarecomposition.ApplicationProfile)
 
 	// if we have an application profile that is marked as complete and completed, we do not allow any updates
-	if oldAP.Annotations[helpers.CompletionMetadataKey] == helpers.Complete && oldAP.Annotations[helpers.StatusMetadataKey] == helpers.Completed {
+	if oldAP.Annotations[helpers.CompletionMetadataKey] == helpers.Complete && oldAP.Annotations[helpers.StatusMetadataKey] == helpers.Completed ||
```

Comment:
can we extract this logic in a utility func that is shared between both strategies AND unit tested?

Reply:
this is becoming harder to maintain